### PR TITLE
Expose SecurityDescriptor as protected

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Classes derived from ObjectSecurity depended on seeing the _securityDescriptor field, which is marked internal. Internal visibility no longer works as types have split into different assemblies and is causing failures.

This change also ran the UpdateVSConfigurations target on the changed libraries.

Fixes: #34151

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Fyi: It is a little strange that this type ended up in `System.IO.FileSystemSecurity` as it has nothing to do with IO. It's only used for Active Directory.

File: src/System.Security.AccessControl/src/System/Security/AccessControl/ObjectSecurity.cs [hunk 1]
```diff
@@ -147,6 +143,11 @@ private void UpdateWithNewSecurityDescriptor( RawSecurityDescriptor newOne, Acce
 
         #region Protected Properties and Methods
 
+        /// <summary>
+        /// Gets the security descriptor for this instance.
+        /// </summary>
+        protected CommonSecurityDescriptor SecurityDescriptor => _securityDescriptor;
```
Comment (stephentoub) [inline src/System.Security.AccessControl/src/System/Security/AccessControl/ObjectSecurity.cs]: Nit: can _securityDescriptor be made readonly?

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 2]
```diff
@@ -19,6 +19,15 @@ public class DirectoryObjectSecurityTests
         private const int WriteAccessMask = 0x02;
         private const int ReadAttributeAccessMask = 0x80;
 
+        [Fact]
+        public void SetCustomDescriptor_Success()
+        {
+            // https://github.com/dotnet/corefx/issues/34151
+            var customObjectSecurity = new CustomDirectoryObjectSecurity();
+            customObjectSecurity.SetSecurityDescriptorSddlForm("D:PAI(A;OICI;0x1200a9;;;BU)");
+            customObjectSecurity.GetAccessRules(true, true, typeof(SecurityIdentifier));
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Can we assert anything?  Validate what GetAccessRules returns?

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 3]
```diff
@@ -19,6 +19,15 @@ public class DirectoryObjectSecurityTests
         private const int WriteAccessMask = 0x02;
         private const int ReadAttributeAccessMask = 0x80;
 
+        [Fact]
+        public void SetCustomDescriptor_Success()
+        {
+            // https://github.com/dotnet/corefx/issues/34151
```
Comment (bartonjs) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: What is the comment supposed to be saying to me?

Comment (JeremyKuhne) [reply src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: That there is a related issue for this test. I can spell it out.

File: src/System.Security.AccessControl/src/System/Security/AccessControl/ObjectSecurity.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Security.AccessControl/src/System/Security/AccessControl/ObjectSecurity.cs]: Sure

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: I was trying to focus on the break here. I'll break out what this SDDL is and validate it.

**Comment by JeremyKuhne** [comment]
Linux arm64_Release timed out deploying. This is in Windows only code. I've followed up on the outstanding comments.

**Pull request merged**